src/components/sharing/SharingSearchSelect.tsx

```diff
@@ -0,0 +1,126 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { IconUser16, IconUserGroup16 } from '@dhis2/ui'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { Sharing } from '../../types/generated'
+import { SearchableSingleSelect } from '../SearchableSingleSelect'
+
+const query = {
+    result: {
+        resource: 'sharing/search',
+        params: ({ searchFilter }: Record<string, string>) => ({
+            key: searchFilter,
+        }),
+    },
+}
+
+type SharingUserResult = {
+    id: string
+    name: string
+    displayName: string
+    username: string
+}
+
+type SharingUserGroupResult = {
+    id: string
+    name: string
+    displayName: string
+}
+
+export type SharingSearchResult =
+    | (SharingUserResult & { entity: 'users' })
+    | (SharingUserGroupResult & {
+          entity: 'userGroups'
+      })
+
+type SharingSearchResponse = {
+    result: {
+        users: SharingUserResult[]
+        userGroups: SharingUserGroupResult[]
+    }
+}
+
+type SharingSearchSelectProps = {
+    onChange: (value: SharingSearchResult) => void
+}
+
+export const SharingSearchSelect = ({ onChange }: SharingSearchSelectProps) => {
+    const [selected, setSelected] = useState<string>('')
+
+    const { data, refetch, error, loading } =
+        useDataQuery<SharingSearchResponse>(query, {
+            variables: { searchFilter: '' },
+        })
+
+    const formattedData: SharingSearchResult[] = useMemo(() => {
+        if (!data?.result) {
+            return []
+        }
+
+        const users = data.result.users.map(
+            (user) =>
+                ({
+                    ...user,
+                    entity: 'users',
+                    label: user.displayName,
+                    value: user.id,
+                } as const)
+        )
+        const userGroups = data.result.userGroups.map(
+            (group) =>
+                ({
+                    ...group,
+                    label: group.displayName,
+                    entity: 'userGroups',
```

Comment:
Fix: remove label and value or fix so that this type can be passed directly to select